scripts/e2e_test/e2e_test.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+
+require 'tmpdir'
+require 'json'
+
+class TestRunner
+  NUM_RUNS_TO_KEEP = 5
+  PROJECT_ROOT = File.expand_path(__dir__ + '/../..') 
+  RUNS_DIR = File.expand_path(__dir__ + '/runs')
+  CYPRESS_DIR = File.join(PROJECT_ROOT, 'packages/cli/cypress')
+
+  E2E_CONFIG = [
```

Comment:
in the future we could parallelize these if we had some better reporting/test output capture